eclass/pam.eclass

```diff
@@ -171,26 +171,25 @@ pamd_mimic_system() {
 pamd_mimic() {
 	[[ $# -lt 3 ]] && die "pamd_mimic requires at least three arguments"
 
-	if has pam ${IUSE} && ! use pam; then
+	if in_iuse pam && ! use pam; then
 		return 0;
 	fi
 
 	dodir /etc/pam.d
-	pamdfile=${D}/etc/pam.d/$2
+	local pamdfile="${ED}/etc/pam.d/$2"
 	echo -e "# File autogenerated by pamd_mimic in pam eclass\n\n" >> \
-		$pamdfile
+		"${pamdfile}" || die
 
-	originalstack=$1
-	authlevels="auth account password session"
+	local authlevels="auth account password session"
 
-	mimic="\tsubstack\t\t${originalstack}"
+	local mimic="\tsubstack\t\t$1"
 
 	shift; shift
 
 	while [[ -n $1 ]]; do
-		has $1 ${authlevels} || die "unknown level type"
+		has $1 "${authlevels}" || die "unknown level type"
```

Comment:
This change doesn't look right. AFAICS you want word-splitting on `${authlevels}`.